packages/flutter_tools/lib/src/fuchsia/fuchsia_build.dart

```diff
@@ -85,10 +85,7 @@ Future<void> _genSnapshot(
   final String appName = fuchsiaProject.project.manifest.appName;
   final String dilPath = globals.fs.path.join(outDir, '$appName.dil');
 
-  final String vmSnapshotData = globals.fs.path.join(outDir, 'vm_data.aotsnapshot');
-  final String vmSnapshotInstructions = globals.fs.path.join(outDir, 'vm_instructions.aotsnapshot');
-  final String snapshotData = globals.fs.path.join(outDir, 'data.aotsnapshot');
-  final String snapshotInstructions = globals.fs.path.join(outDir, 'instructions.aotsnapshot');
+  final String elf = globals.fs.path.join('elf.aotsnapshot');
```

Comment:
Should this be:
```dart
globals.fs.path.join(outDir, 'elf.aotsnapshot');
```

Reply:
Done. I'm a little confused how `flutter run` worked in that case: don't we use AOT for `flutter run --release`?

Reply:
I can see actually that before this fix there was no instance of `io.flutter.product_runner_aot` in `ps`, and now there is. Does it fall back to a dill if it can't load the AOT snapshot?

Reply:
I don't think it will do a fallback, and I'm pretty sure there wouldn't have been a dill in the package for it to fall back to. Was there anything interesting in the system logs before the fix when there was no product runner instance?

Reply:
The actual issue is that I wasn't removing the `flutter_tools.snapshot`, so my code wasn't running. After fixing that, I can confirm that the app does not launch before this fix and does launch after.